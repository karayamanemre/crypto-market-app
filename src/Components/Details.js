import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCategory } from '../Redux/Category/categories';
import styles from './Styles/Details.module.css';

const Details = () => {
  const dispatch = useDispatch();
  const params = useParams();
  const id = params.coinId;
  const details = useSelector((state) => state.detail);
  useEffect(() => {
    dispatch(fetchCategory(id));
  }, [dispatch, id]);
  return (
    <section>
      <div className={styles.container}>
        <h1>{details.name}</h1>
        <span>
          Rank #{
          details.market_cap_rank}</span>
        {details.image ? (
          <img src={details.image.small} alt="coin logo" />
        ) : null}
        {details.symbol ? <p>{details.symbol.toUpperCase()}</p> : null}
        {details.market_data ? (
          <h1>
            $
            {details.market_data.current_price.usd.toLocaleString()}
          </h1>
        ) : null}
        <ul>
          <li>
            <h4>24 Hour Change:</h4>
            {details.market_data?.price_change_percentage_24h_in_currency
              ? details.market_data.price_change_percentage_24h_in_currency.usd.toFixed(
                1,
              )
              : null}
              {' '}
            %
          </li>
          <li>
            <h4>24 Hour Low:</h4>
            {details.market_data?.low_24h ? (
              <p>
                $
                {details.market_data.low_24h.usd.toLocaleString()}
              </p>
            ) : null}
          </li>
          <li>
            <h4>24 Hour High:</h4>
            {details.market_data?.high_24h ? (
              <p>
                $
                {details.market_data.high_24h.usd.toLocaleString()}
              </p>
            ) : null}
          </li>
          <li>
            <h4>Market Cap:</h4>
            {details.market_data?.market_cap ? (
              <p>
                $
                {details.market_data.market_cap.usd.toLocaleString()}
              </p>
            ) : null}
          </li>
          <li>
            <h4>Circulating Supply:</h4>
            {details.market_data?.circulating_supply ? (
              <p>
                $
                {details.market_data.circulating_supply.toLocaleString()}
              </p>
            ) : null}
          </li>
        </ul>
      </div>
    </section>
  );
};

export default Details;
