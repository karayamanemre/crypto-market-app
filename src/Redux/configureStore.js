import { createStore, combineReducers } from 'redux';
import currencyReducer from './Currencies/currencies';
import categoryReducer from './Category/categories';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { applyMiddleware } from 'redux';

const rootReducer = combineReducers({
  coins: currencyReducer,
  detail: categoryReducer,
});

export const store = createStore(rootReducer, applyMiddleware(thunk, logger));
