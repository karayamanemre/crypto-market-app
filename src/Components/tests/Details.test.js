import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../Redux/configureStore';
import Details from '../Details';

describe('Details page test', () => {
  test('render details page', () => {
    const view = render(
      <Provider store={store}>
        <Details />
      </Provider>,
    );
    expect(view).toMatchSnapshot();
  });
});
