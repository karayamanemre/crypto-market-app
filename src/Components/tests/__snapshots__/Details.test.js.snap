// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details page test render details page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="container"
        >
          <h1 />
          <span>
            Rank # 
          </span>
          <ul>
            <li>
              <h4>
                24 Hour Change:
              </h4>
               %
            </li>
            <li>
              <h4>
                24 Hour Low:
              </h4>
            </li>
            <li>
              <h4>
                24 Hour High:
              </h4>
            </li>
            <li>
              <h4>
                Market Cap:
              </h4>
            </li>
            <li>
              <h4>
                Circulating Supply:
              </h4>
            </li>
          </ul>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="container"
      >
        <h1 />
        <span>
          Rank # 
        </span>
        <ul>
          <li>
            <h4>
              24 Hour Change:
            </h4>
             %
          </li>
          <li>
            <h4>
              24 Hour Low:
            </h4>
          </li>
          <li>
            <h4>
              24 Hour High:
            </h4>
          </li>
          <li>
            <h4>
              Market Cap:
            </h4>
          </li>
          <li>
            <h4>
              Circulating Supply:
            </h4>
          </li>
        </ul>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
